liblangutil/SourceLocation.h

```diff
@@ -102,6 +103,7 @@ struct SourceLocation
 	int start = -1;
 	int end = -1;
 	std::shared_ptr<std::string const> sourceName;
+	std::optional<int> source = std::nullopt;
```

Comment:
`source` sounds like something that would contain the source code.
```suggestion
	std::optional<int> sourceIndex = std::nullopt;
```

Reply:
Is this redundancy required? Can't we just store the source name?

Reply:
Yeah that would be perfect. The problem is that we don't have the source name available in the json, only the source index.

Reply:
But maybe we could just change the structure of the assembly json. But I guess that would be breaking.